proposal.rst

```diff
@@ -180,39 +180,57 @@ Motivations
 
-It is interesting to me to benchmark ranking algorithms on big databases.
+Information Retrieval is interesting to me as application of statistics. This project allows me to broaden horizons with practical examples. Also programming practice and communicating with mentors are very important to me.
 
 **Who will benefit from your project and in what ways?**
 
 .. For example, think about the likely user-base, what they currently have to
 .. do and how your project will improve things for them.
 
-Project I want to do assumes that there will be tool for evaluating Letor on different datasets and compare its performance, so it useful for users wich want to understand Xapian abilities.
+Project I want to do assumes that there will be tool for evaluating Letor on different datasets and compare its performance. If it is completed successfully, there free suite with several ranking algorithms and features will available for users. It is useful not only for future developers, but for participants IR contests like TREC. Also it is possible to implement automated configuration choose using score classes.  
 
 Project Details
 ---------------
 
-.. Please go into plenty of detail in this section.
-
 **Describe any existing work and concepts on which your project is based.**
 
-There are several ranking algorithms with pretty same api implemented in Letor. So I think manager class should be developed to evaluate ranking and logging results. It will better to integrate it with testing classes, because it makes possible to develop suite for perfomance rating, wich will useful not only for ranking, but for all test cases. 
+There are several ranking algorithms with pretty same api implemented in Letor. So I think manager class should be developed to evaluate ranking and logging results. It will better to integrate it with Letor classes. Additional task is to improve perftest suite by running it on real data, wich will useful not only for ranking benchmark, but for all test cases. More detailed:
+
+* Implement class for preparing data, i.e. download it and convert to suitable form. The most convenient dataset is `FIRE <http://fire.irsi.res.in/fire/static/data>`_ and `TREC <https://trec.nist.gov/data/t2002_filtering.html>`_. It is hard to fully generalize this case, so I prefer to implement parser for each dataset:
+
+	* FIRE queries stored in inappropriate format, they should be transformed to use them in Xapian; It will possible to use English language corpus, because Eglish stemmer exists in Xapian.
+
+	* TREC dataset has big enough amount of documents, it is possible to combine some of them during pasing to create more complex queries.
+
+* Implement evaluator. According to xapian-letor/tests/api_letor.cc it is no big difference in applying rankers to data, it should just apply algorithms and return scores, so it could be another one class in xapian/letor. It would be nice to be able to call it same way as make check.
```

Comment:
How can you make this as simple as `make check`? Will there be a standard place to store the source corpus, queries, and qrels? What do you actually want people to be able to run? Writing out the steps that someone will have to take to use this will help you break down what you need to implement, as well as ensure we understand clearly what you are proposing.